server/paymentProviders/transferwise/index.ts

```diff
@@ -73,56 +79,60 @@ async function payExpense(connectedAccount, payoutMethod, expense): Promise<any>
+  const connectedAccount =
+    host.connectedAccount?.service === 'transferwise'
+      ? host.connectedAccount
+      : await models.ConnectedAccount.findOne({
+          where: { service: 'transferwise', CollectiveId: host.id, deletedAt: null },
+        });
   if (!connectedAccount) {
     throw new Error('Host is not connected to Transferwise');
   }
 
-  const quote = await transferwise.createQuote(connectedAccount.token, {
-    profileId: connectedAccount.data.id,
-    sourceCurrency: host.currency,
-    targetCurrency: currency,
-    targetAmount: 100,
-  });
-  const requiredFields = await transferwise.getAccountRequirements(connectedAccount.token, quote.id);
-  cache.set(cacheKey, requiredFields, 24 * 60 * 60 /* a whole day and we could probably increase */);
-  return requiredFields;
+  const pairs = await transferwise.getCurrencyPairs(connectedAccount.token);
+  const source = pairs.sourceCurrencies.find(sc => sc.currencyCode === host.currency);
+  const currencies = source.targetCurrencies
+    .filter(c => !blackListedCurrencies.includes(c.currencyCode))
+    .map(c => ({ code: c.currencyCode, minInvoiceAmount: c.minInvoiceAmount }));
+  cache.set(cacheKey, currencies, 24 * 60 * 60 /* a whole day and we could probably increase */);
+  return currencies;
 }
 
-async function getAvailableCurrencies(host: any): Promise<any> {
-  const cacheKey = `transferwise_available_currencies_${host.id}`;
+async function getRequiredBankInformation(host: any, currency: string): Promise<any> {
+  const cacheKey = `transferwise_required_bank_info_${host.id}_to_${currency}`;
   const fromCache = await cache.get(cacheKey);
   if (fromCache) {
     return fromCache;
   }
 
-  const connectedAccount = await models.ConnectedAccount.findOne({
-    where: {
-      service: 'transferwise',
-      CollectiveId: host.id,
-      deletedAt: null,
-    },
-  });
+  const connectedAccount =
+    host.connectedAccount?.service === 'transferwise'
+      ? host.connectedAccount
+      : await models.ConnectedAccount.findOne({
+          where: { service: 'transferwise', CollectiveId: host.id, deletedAt: null },
+        });
   if (!connectedAccount) {
     throw new Error('Host is not connected to Transferwise');
   }
 
-  const pairs = await transferwise.getCurrencyPairs(connectedAccount.token);
-  const source = pairs.sourceCurrencies.find(sc => sc.currencyCode === host.currency);
-  const currencies = source.targetCurrencies
-    .filter(c => !blackListedCurrencies.includes(c.currencyCode))
-    .map(c => c.currencyCode);
-  cache.set(cacheKey, currencies, 24 * 60 * 60 /* a whole day and we could probably increase */);
-  return currencies;
+  const currencyInfo = find(await getAvailableCurrencies(host), { code: currency });
+
+  const quote = await transferwise.createQuote(connectedAccount.token, {
+    profileId: connectedAccount.data.id,
+    sourceCurrency: host.currency,
+    targetCurrency: currency,
+    sourceAmount: currencyInfo.minInvoiceAmount * 20,
```

Comment:
Won't this will crash if we call `getRequiredBankInformation` with a currency that is not supported?

Reply:
It will throw, but the currencies available are also being generated by TransferWise.
I already blacklisted a few currencies that could present specific restrictions, but if we're aiming to be 100% safe I think we could whitelist just a few currencies we're sure it works.

Reply:
It's always better to gracefully fail, we could just add above:

```ts
const currencyInfo = find(await getAvailableCurrencies(host), { code: currency });
if (!currencyInfo) {
  throw new Error('This currency is not supported');
}
```